dns.go

```diff
@@ -231,22 +224,6 @@ func (api *API) UpdateDNSRecord(ctx context.Context, rc *ResourceContainer, para
 
 	params.Name = toUTS46ASCII(params.Name)
 
-	// Populate the record name from the existing one if the update didn't
-	// specify it.
-	if params.Name == "" || params.Type == "" {
```

Comment:
i suspect this was from a time when updating DNS records was a PUT and these values were required to prevent broken resources if the `Name` or `Type` was omitted.

Reply:
Yes, I think so. (Too lazy to check the git history.)